### ISSUE TITLE
Add subaccount resource

### ISSUE THREAD

**Issue by cen1** [SparkPost/java-sparkpost#50]
Unless I am missing something there seems to be no support for managing subaccounts. Would be nice to add so we don't have to write seperate clients.

**Comment by yepher** [comment]
Sub accounts should work with this library. Can you give me an example so I can make an example for you?

**Comment by cen1** [comment]
Let's say how to create a new subaccount.

**Comment by yepher** [comment]
Oh, I see what you mean now. Yes, that does need added. I would be happy to merge a pull request if you are interested in working on it.

Otherwise I will work on it in a week or so.

**Comment by cen1** [comment]
I might do a pull request but I make no promises. If there is none in a week go ahead with the implementation. Thanks.

**Comment by cen1** [comment]
Further, is it possible to add X-MSYS-SUBACCOUNT header in any way? Looking at the RestConnection implementation there seems to be no way to manipulate HttpURLConnection so my guess is a no.

If you can give me that example for calling the API as a subaccount I'd appreciate it.

**Comment by yepher** [comment]
I thought that was on a branch but I don't see it. I will add that ability.

**Comment by scarlier** [comment]
any progress on this issue?

**Comment by yepher** [comment]
I am going to start implementing this now. My goal is to not break backward compatibility (somehow).

**Comment by yepher** [comment]
I've added an initial implementation for sub account support to the "feature/subaccount" branch.

Usage is like this:

```
IRestConnection connection = new RestConnection(this.client, getEndPoint());
connection.setSubAccountId("1234");
```

Anything that connection is used for will set the `X-MSYS-SUBACCOUNT` header and the request.

**Comment by scarlier** [comment]
would be nice to see this update on maven

**Comment by yepher** [comment]
This is released in version 0.22
